release-testing/prepare.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+. ./store.sh
+. ./executors.sh
+
+function deploy_pg {
+  helm install --namespace "$(store_get TEST_NAMESPACE)" "$(store_get PG_RELEASE_NAME)" \
+   --set postgresqlPassword="$(store_get DB_PASSWORD)" \
+   --set persistence.enabled="false" \
+   stable/postgresql
+}
+
+function deploy_conjur {
+  helm install --namespace "$(store_get TEST_NAMESPACE)" "$(store_get CONJUR_RELEASE_NAME)" \
+    --set dataKey="$(store_get DATA_KEY)" \
+    --set account="$(store_get CONJUR_ACCOUNT)" \
+    --set image.tag="$(store_get CONJUR_OSS_RELEASE_VERSION)" \
+    --set image.pullPolicy="Always" \
+    --set authenticators="authn-k8s/$(store_get AUTHENTICATOR_ID)\,authn" \
+    --set databaseUrl="postgres://postgres:$(store_get DB_PASSWORD)@$(store_get PG_RELEASE_NAME)-postgresql.$(store_get TEST_NAMESPACE).svc.cluster.local/postgres" \
+    "https://github.com/cyberark/conjur-oss-helm-chart/releases/download/v$(store_get HELM_CHART_RELEASE_VERSION)/conjur-oss-$(store_get HELM_CHART_RELEASE_VERSION).tgz"
+}
+
+function register_conjur_pod() {
+  local CONJUR_POD_NAME=$(kubectl --namespace "$(store_get TEST_NAMESPACE)" get pods \
+                   -l "app=conjur-oss,release=$(store_get CONJUR_RELEASE_NAME)" \
+                   -o jsonpath="{.items[0].metadata.name}")
+
+  store_set CONJUR_POD_NAME "${CONJUR_POD_NAME}"
+}
+
+function register_conjur_client_pod() {
+  local CONJUR_CLIENT_POD_NAME="conjur-client-pod"
+
+  # start the CLI pod
+  kubectl --namespace "$(store_get TEST_NAMESPACE)" run "${CONJUR_CLIENT_POD_NAME}" \
+   --restart='Never' \
+   --image cyberark/conjur-cli:5 \
+   --command -- sleep infinity
+
+  # wait for CLI pod to be ready
+  kubectl --namespace "$(store_get TEST_NAMESPACE)" \
+   wait --for=condition=ready "pod/${CONJUR_CLIENT_POD_NAME}" --timeout 150s
+
+  # login to conjur
+  kubectl --namespace "$(store_get TEST_NAMESPACE)" \
+    exec -i "${CONJUR_CLIENT_POD_NAME}" -- \
+     bash -xce "
+yes yes | conjur init -u '$(store_get CONJUR_URL)' -a '$(store_get CONJUR_ACCOUNT)'
+
+# API key here is the key that creation of the account provided you in step #2
+conjur authn login -u admin -p '$(store_get CONJUR_ADMIN_API_KEY)'
+
+# Check that you are identified as the admin user
+conjur authn whoami
+"
+
+  store_set CONJUR_CLIENT_POD_NAME "${CONJUR_CLIENT_POD_NAME}"
+}
+
+function register_conjur_admin_key() {
+  local CONJUR_ADMIN_API_KEY=$(exec_conjur \
+   conjurctl role retrieve-key "$(store_get CONJUR_ACCOUNT):user:admin")
+
+  store_set CONJUR_ADMIN_API_KEY "${CONJUR_ADMIN_API_KEY}"
```

Comment:
I think it's okay to store this API key... given that this Conjur cluster is an ephemeral cluster used for testing?

Reply:
I think it's OK